src/creatures/players/cast/cast_viewer.cpp

```diff
@@ -0,0 +1,1533 @@
+			it.first->sendCreatureLight(creature);
+		}
+	}
+}
+
+void CastViewer::sendCreatureWalkthrough(std::shared_ptr<Creature> creature, bool walkthrough) {
+	if (m_owner) {
+		m_owner->sendCreatureWalkthrough(creature, walkthrough);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendCreatureWalkthrough(creature, walkthrough);
+		}
+	}
+}
+
+void CastViewer::sendCreatureShield(std::shared_ptr<Creature> creature) {
+	if (m_owner) {
+		m_owner->sendCreatureShield(creature);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendCreatureShield(creature);
+		}
+	}
+}
+
+void CastViewer::sendContainer(uint8_t cid, std::shared_ptr<Container> container, bool hasParent, uint16_t firstIndex) {
+	if (m_owner) {
+		m_owner->sendContainer(cid, container, hasParent, firstIndex);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendContainer(cid, container, hasParent, firstIndex);
+		}
+	}
+}
+
+void CastViewer::sendInventoryItem(Slots_t slot, std::shared_ptr<Item> item) {
+	if (m_owner) {
+		m_owner->sendInventoryItem(slot, item);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendInventoryItem(slot, item);
+		}
+	}
+}
+
+void CastViewer::sendCancelMessage(const std::string &msg) const {
+	if (m_owner) {
+		m_owner->sendTextMessage(TextMessage(MESSAGE_FAILURE, msg));
+
+		for (const auto &it : m_viewers) {
+			it.first->sendTextMessage(TextMessage(MESSAGE_FAILURE, msg));
+		}
+	}
+}
+
+void CastViewer::sendCancelTarget() const {
+	if (m_owner) {
+		m_owner->sendCancelTarget();
+
+		for (const auto &it : m_viewers) {
+			it.first->sendCancelTarget();
+		}
+	}
+}
+
+void CastViewer::sendCancelWalk() const {
+	if (m_owner) {
+		m_owner->sendCancelWalk();
+
+		for (const auto &it : m_viewers) {
+			it.first->sendCancelWalk();
+		}
+	}
+}
+
+void CastViewer::sendChangeSpeed(std::shared_ptr<Creature> creature, uint32_t newSpeed) const {
+	if (m_owner) {
+		m_owner->sendChangeSpeed(creature, newSpeed);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendChangeSpeed(creature, newSpeed);
+		}
+	}
+}
+
+void CastViewer::sendCreatureHealth(std::shared_ptr<Creature> creature) const {
+	if (m_owner) {
+		m_owner->sendCreatureHealth(creature);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendCreatureHealth(creature);
+		}
+	}
+}
+
+void CastViewer::sendDistanceShoot(const Position &from, const Position &to, unsigned char type) const {
+	if (m_owner) {
+		m_owner->sendDistanceShoot(from, to, type);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendDistanceShoot(from, to, type);
+		}
+	}
+}
+
+void CastViewer::sendCreatePrivateChannel(uint16_t channelId, const std::string &channelName) {
+	if (m_owner) {
+		m_owner->sendCreatePrivateChannel(channelId, channelName);
+	}
+}
+
+void CastViewer::sendIcons(uint32_t icons) const {
+	if (m_owner) {
+		m_owner->sendIcons(icons);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendIcons(icons);
+		}
+	}
+}
+
+void CastViewer::sendMagicEffect(const Position &pos, uint8_t type) const {
```

Comment:
It should be 16 bits.
```suggestion
void CastViewer::sendMagicEffect(const Position &pos, uint16_t type) const {
```